apps/test/util/testUtils.js

```diff
@@ -278,11 +285,16 @@ export function throwOnConsoleWarnings() {
         console.log('Unexpected call to console.warn: ' + msg);
         console.log(e.stack);
       }
-      throw new Error(msg);
+      // Store error so we can throw in after. This will ensure we hit a failure
+      // even if message was originally thrown in async code
+      firstErrorMessage = firstErrorMessage || msg;
```

Comment:
I'm not sure we could test this within the same suite, now that we're throwing _outside_ of the respective test.  I'm fine with that, as long as we're confident that it's working as designed.

I might make some changes to this alternative implementation though.

1. We should probably use `beforeEach` and `afterEach` instead of `before` and `after` so the error gets thrown as close to the problematic test as possible.
2. Maybe we should actually pass out the error object with the stack instead of logging inline here and passing a string out as `firstErrorMessage`.  Might be a little less convoluted.